### PR TITLE
[libc] android atest compat

### PR DESCRIPTION
**Pull request by nickdesaulniers**
See individual commit messages in GH PR Review.

These changes slighly modify the output of the unittests so that they better
match GTest, so that utilities that parse the expected output from GTest (such
as Android's unit test harness) can read the output from our unit tests.

This allows our unit tests to be run on Android devices.

- **libc: UnitTest: print number of tests being run**
- **libc: UnitTest: basic command line flag support**

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
cc @enh-google

**Review by enh-google — APPROVED** [review]

**Review by enh-google — APPROVED** [review]

**Review by enh-google — APPROVED** [review]

**Review by enh-google — APPROVED** [review]

File: libc/test/UnitTest/LibcTest.cpp [hunk 1]
```diff
@@ -127,19 +127,37 @@ void Test::addTest(Test *T) {
...
   int FailCount = 0;
   for (Test *T = Start; T != nullptr; T = T->Next) {
     const char *TestName = T->getName();
-    cpp::string StrTestName(TestName);
-    constexpr auto GREEN = "\033[32m";
-    constexpr auto RED = "\033[31m";
-    constexpr auto RESET = "\033[0m";
-    if ((TestFilter != nullptr) && (StrTestName != TestFilter)) {
+
+    if (Options.TestFilter && cpp::string{TestName} != Options.TestFilter) {
```
Comment (lntue) [inline libc/test/UnitTest/LibcTest.cpp:155]: nit: why `cpp::string{..}` but not `cpp::string(...)`?

File: libc/test/UnitTest/LibcTest.h [hunk 2]
```diff
@@ -100,20 +100,30 @@ bool test(RunContext *Ctx, TestCond Cond, ValType LHS, ValType RHS,
 
 } // namespace internal
 
+struct TestOptions {
+  // If set, then just this one test from the suite will be runt.
```
Comment (michaelrj-google) [inline libc/test/UnitTest/LibcTest.h:104]: nit: runt -> run

**Review by michaelrj-google — APPROVED** [review]
LGTM with a nit

File: libc/test/UnitTest/LibcTest.cpp (hunk 1 above)
Comment (nickdesaulniers) [reply libc/test/UnitTest/LibcTest.cpp:155]: :shrug: 

Looks like LLVM style guide says not to do what I did, will change

https://llvm.org/docs/CodingStandards.html#do-not-use-braced-initializer-lists-to-call-a-constructor

Comment (nickdesaulniers) [reply libc/test/UnitTest/LibcTest.cpp]: 35a661687a03

File: libc/test/UnitTest/LibcTest.h (hunk 2 above)
Comment (nickdesaulniers) [reply libc/test/UnitTest/LibcTest.h]: 61124058d2ff

**Pull request merged**